### PR TITLE
feat(cli): add task delete command

### PR DESCRIPTION
**Pull request by bvtujo**
This PR does the following:
* Adds `task delete` as a new command
* Adds a step in `app delete` during which all existing tasks in the application's environments are deleted
* Rephrases the e2e test for tasks to use the new command on teardown
* Adds documentation of the new command. 

Depends on #1840 being merged in. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: site/content/docs/commands/task-delete.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+## What does it do?
+`copilot task delete` removes the resources created by a one-off task.
+
+Generally, the steps involved in `task delete` are:
+
+1. Stop all running instances of the task.
+2. Empty the ECR repository for the task.
+3. Delete the CloudFormation stack, including the ECR repo, log group, and ECS Task Definition.
```
Comment (efekarakus) [inline site/content/docs/commands/task-delete.md]: I think we can remove this info, it seems like it's internal details that we're surfacing to customers

File: site/content/docs/commands/task-delete.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# task delete
+```
+$ copilot task delete
+```
+
+## What does it do?
+`copilot task delete` removes the resources created by a one-off task.
```
Comment (efekarakus) [inline site/content/docs/commands/task-delete.md]: what do you think of this:
```suggestion
`copilot task delete` stops running instances of the task, and deletes associated resources.
```

File: site/content/docs/commands/task-delete.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Generally, the steps involved in `task delete` are:
+
+1. Stop all running instances of the task.
+2. Empty the ECR repository for the task.
+3. Delete the CloudFormation stack, including the ECR repo, log group, and ECS Task Definition.
+
+!!!info
+    1. Tasks created with versions of Copilot earlier than v1.2.0 cannot be stopped by `copilot task delete`. Customers using tasks launched with earlier versions should manually stop any running tasks via the ECS console after running the command. 
+    2. If you are using the `--default` flag, you cannot also specify the `--app` or `--env` flags.
```
Comment (efekarakus) [inline site/content/docs/commands/task-delete.md]: we can remove this I think, they'll get an error when they run the cmd.

File: site/content/docs/commands/task-delete.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+`copilot task delete` removes the resources created by a one-off task.
+
+Generally, the steps involved in `task delete` are:
+
+1. Stop all running instances of the task.
+2. Empty the ECR repository for the task.
+3. Delete the CloudFormation stack, including the ECR repo, log group, and ECS Task Definition.
+
+!!!info
+    1. Tasks created with versions of Copilot earlier than v1.2.0 cannot be stopped by `copilot task delete`. Customers using tasks launched with earlier versions should manually stop any running tasks via the ECS console after running the command.
```
Comment (efekarakus) [inline site/content/docs/commands/task-delete.md]: What was the reason for this?

**Review by efekarakus — COMMENTED** [review]
Looks great! just some documentation suggestions

File: site/content/docs/commands/task-delete.md (hunk 4 above)
Comment (bvtujo) [reply site/content/docs/commands/task-delete.md]: Previously, the call in `task run` to `ecs.RunTask` did not have the `PropagateTags: aws.String(ecs.PropagateTagsTaskDefinition)` property set so the tasks cannot be reliably identified as Copilot-initiated.

Comment (bvtujo) [reply site/content/docs/commands/task-delete.md]: The side effect would be the ability to stop tasks from a one-off task definition not started by copilot.

Comment (efekarakus) [reply site/content/docs/commands/task-delete.md]: ooh yeaa gotcha, thanks for the clarification

**Review by efekarakus — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]

**Comment by bvtujo** [comment]
Revised E2E test passes. 
```
...
    /github.com/aws/copilot-cli/e2e/task/task_test.go:128
------------------------------
✔ Stopped all running tasks in family e2e-task-1611705473.
✔ Emptied ECR repositories for task e2e-task-1611705473.
✔ Deleted resources of task e2e-task-1611705473.
Recommended follow-up actions:
✔ Deleted environment test from application e2e-task-1611705473.
✔ Cleaned up deployment resources.
✔ Deleted application resources.
✔ Deleted application configuration.
✔ Deleted local .workspace file.

Ran 7 of 11 Specs in 768.633 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 4 Skipped
PASS
```

**Pull request merged**